scripts/esbuild/testing.ts

```diff
@@ -0,0 +1,141 @@
+import type { BuildOptions as ESBuildOptions, Plugin } from 'esbuild';
+import fs from 'fs-extra';
+import { join } from 'path';
+
+import { copyTestingInternalDts } from '../bundles/testing';
+import { getBanner } from '../utils/banner';
+import type { BuildOptions } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { getBaseEsbuildOptions, getEsbuildAliases, getEsbuildExternalModules, runBuilds } from './util';
+
+const EXTERNAL_TESTING_MODULES = [
+  'assert',
+  'buffer',
+  'child_process',
+  'console',
+  'constants',
+  'crypto',
+  'fs',
+  '@jest/core',
+  'jest-cli',
+  'jest',
+  'expect',
+  '@jest/reporters',
+  'jest-environment-node',
+  'jest-message-id',
+  'jest-runner',
+  'net',
+  'os',
+  'path',
+  'process',
+  'puppeteer',
+  'puppeteer-core',
+  'readline',
+  'rollup',
+  '@rollup/plugin-commonjs',
+  '@rollup/plugin-node-resolve',
+  'stream',
+  'tty',
+  'url',
+  'util',
+  'vm',
+  'yargs',
+  'zlib',
+];
+
+export async function buildTesting(opts: BuildOptions) {
+  const inputDir = join(opts.buildDir, 'testing');
+  const sourceDir = join(opts.srcDir, 'testing');
```

Comment:
What do you think about adding an `fs.emptyDir` call before we starting writing files, with `opts.output.testingDir` as the argument, so we can be sure the esbuild-based build is 'clean' and we don't have files from previous builds hanging around?

Reply:
we do that here, for instance: 

https://github.com/ionic-team/stencil/blob/276cc7370b6bf22326b74f7a314a67a27f23cf7b/scripts/esbuild/sys-node.ts#L17-L18

Reply:
added

```ts
await fs.emptyDir(opts.output.testingDir);
```